### PR TITLE
Do not update if there is a save pending

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
This attempts to solve another round synchronization issue - the suspected culprits are the debouncers we are using for both saving-to-db and update-from-db going out of sync with each other. The primary reason for debouncers here is because we cannot send in a stream of updates across REST API without jamming-up every one else, so to resolve around that we introduce a couple of heuristics below.

To illustrate the potential sync problem: 
- Action 1: if we move operator-A, then
- Action 2: link operator-A to operator-B
- Action 1 is send to DB, and Action 2 is queued
  - As Action 2 is queued up, the response from Action 1 comes back and partially overwrite what we would've written with Action 2, at this point Action 2 is flushed tot he DB.

Heuristic 1: Do not update if there is a save action pending

Heuristic 2: Do not update if we are dragging

In both cases, we acknowledge there is an update by incrementing the version number, but skip over the fields. The idea is that, for
- heuristic 1: there is a save in progress that will result in a update-from-db, we will use that one 
- heuristic 2: drag-end results in a save, which is heuristic 1

The assumption is that only a single user works on an operator at a time.

Relates to
- https://github.com/DARPA-ASKEM/terarium/issues/4828
- https://github.com/DARPA-ASKEM/terarium/issues/4406

### Testing
This should remedy strange jerkiness/flickers, as well as weird connection states between operator ports.

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/tera-workflow.vue [hunk 1]
```diff
@@ -300,28 +305,34 @@ const teraOperatorRefs = ref();
...
 // eslint-disable-next-line
 const _updateWorkflow = (event: any) => {
-	if (event.data.id !== wf.value.getId()) return;
-
-	// console.log('update workflow', event.data);
-	wf.value.update(event.data as Workflow);
+	if (event.data.id !== wf.value.getId()) {
+		return;
+	}
+	wf.value.update(event.data as Workflow, pendingSave, isDragging);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/tera-workflow.vue:325]: If you do not like one line returns, why not reverse the conditional test?
```suggestion
	if (event.data.id === wf.value.getId()) {
		wf.value.update(event.data as Workflow, pendingSave, isDragging);
	}
```

File: packages/client/hmi-client/src/services/workflow.ts [hunk 2]
```diff
@@ -65,6 +65,28 @@ export class WorkflowWrapper {
 		const updatedNodeMap = new Map<string, WorkflowNode<any>>(updatedWF.nodes.map((n) => [n.id, n]));
 		const updatedEdgeMap = new Map<string, WorkflowEdge>(updatedWF.edges.map((e) => [e.id, e]));
 
+		if (pendingSave || isDragging) {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/workflow.ts:68]: Why not creating a single boolean test with a more appropriate names, or a comment explaining why this test is happening?

**Review by YohannParis — APPROVED** [review]

**Review by shawnyama — APPROVED** [review]

File: packages/client/hmi-client/src/services/workflow.ts (hunk 2 above)
Comment (Tom-Szendrey) [inline packages/client/hmi-client/src/services/workflow.ts:68]: I was not understanding why we would want to update the local version here 
Thank you for your help
May help to have a quick comment explaining this is to allow the user's outdated version to overwrite things prior to pulling

**Review by Tom-Szendrey — APPROVED** [review]
Awesome!

File: packages/client/hmi-client/src/services/workflow.ts (hunk 2 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/services/workflow.ts:68]: will do

**Pull request merged**